build.rs

```diff
@@ -209,14 +209,65 @@ fn build_v8(is_asan: bool) {
     "use_custom_libcxx={}",
     env::var("CARGO_FEATURE_USE_CUSTOM_LIBCXX").is_ok()
   ));
-  gn_args.push(format!(
-    "v8_enable_pointer_compression={}",
-    env::var("CARGO_FEATURE_V8_ENABLE_POINTER_COMPRESSION").is_ok()
-  ));
+
+  let extra_args = {
+    if env::var("CARGO_FEATURE_V8_ENABLE_SANDBOX").is_ok()
+      && env::var("CARGO_FEATURE_V8_ENABLE_POINTER_COMPRESSION").is_ok()
+    {
+      panic!(
+        "Sandbox and pointer compression cannot be enabled at the same time"
+      );
+    }
+
+    if env::var("CARGO_FEATURE_V8_ENABLE_SANDBOX").is_ok() {
+      vec![
+        // Enable pointer compression (along with its dependencies)
+        "v8_enable_sandbox=true",
+        "v8_enable_external_code_space=true", // Needed for sandbox
+        "v8_enable_pointer_compression=true",
+        // Note that sandbox requires shared_ro_heap and verify_heap
+        // to be true/default
+      ]
+    } else {
+      let mut opts = vec![
+        // Disable sandbox
+        "v8_enable_sandbox=false",
+        // Enabling the shared read-only heap comes with a restriction that all
+        // isolates running at the same time must be created from the same snapshot.
+        // This is problematic for Deno, which has separate "runtime" and "typescript
+        // compiler" snapshots, and sometimes uses them both at the same time.
+        //
+        // NOTE FOR FUTURE: Check if this flag even exists anymore as it has likely been
+        // removed
+        "v8_enable_shared_ro_heap=false",
+        // V8 11.6 hardcoded an assumption in `mksnapshot` that shared RO heap
+        // is enabled. In our case it's disabled so without this flag we can't
+        // compile.
+        //
+        // NOTE FOR FUTURE: Check if this flag even exists anymore as it has likely been
+        // removed
+        "v8_enable_verify_heap=false",
```

Comment:
fwiw these flags can be removed per #1788

Reply:
Yeah, I did see the PR there but decided removing these flags would probably be out of scope for a PR that just adds support for v8 sandbox and keeps existing behavior if so. 

My reasoning here is that I’m not sure if deno still relies on having separate runtime and ts snapshots in which case it might be a good idea for deno to just revert the commit which removed the enable shared ro heap flag. In either case, it’s an issue that deno needs to go through and solve versus just removing the dead flags (just my personal opinion here)

Reply:
seems like rusty v8 itself got rid of them so this is fixed too